### PR TITLE
#36043 Fix autocompletion for join conditions

### LINKED ISSUE #36043: New autocomplete is broken for JOINs
### Description

dbeaver_lXrn4QTqge

https://github.com/user-attachments/assets/0394d3df-bd94-4919-bae5-fb8e8f681d8e

### DBeaver Version

DBeaver 24.2.4

### Operating System

windows 10

### Database and driver

_No response_

### Steps to reproduce

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by E1izabeth**
image

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.model.lsm/grammar/SQLStandardParser.g4 [hunk 1]
```diff
@@ -215,7 +215,7 @@ tableSubquery: subquery;
 
 //joins
 crossJoinTerm: CROSS JOIN tableReference;
-naturalJoinTerm: (NATURAL)? (joinType)? JOIN tableReference (joinSpecification)? anyUnexpected??; // (.*?) - for error recovery
+naturalJoinTerm: (NATURAL)? (joinType)? JOIN tableReference (joinSpecification)? anyUnexpected??;//|anyUnexpected); // (.*?) - for error recovery
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model.lsm/grammar/SQLStandardParser.g4]: What was changed?

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.model.lsm/grammar/SQLStandardParser.g4]: Ups

**Review by ShadelessFox — APPROVED** [review]

**Review by HocKu7 — APPROVED** [review]

**Comment by MashaKorax** [comment]
dvdrental.Masha_parser
```
SELECT * FROM
(SELECT * FROM t2) as s2
INNER JOIN
(SELECT * FROM t3) s3
USING (name);
```
https://github.com/user-attachments/assets/0b24616c-81a7-4ade-a4e0-3fd4aee67883

**Comment by E1izabeth** [comment]
Changes are integrated into [another PR](https://github.com/dbeaver/dbeaver/pull/36469) to avoid conflicts

**Pull request merged**